pyproject.toml

```diff
@@ -261,61 +276,145 @@ exclude = [
+    "E",      # pycodestyle (part of default flake8)
+    "W",      # pycodestyle (part of default flake8)
+    "D",      # docstrings, see also numpydoc pre-commit action
+    "N",      # pep8-naming (naming conventions)
+    "A",      # flake8-builtins (prevent shadowing of builtins)
+    "ARG",    # flake8-unused-arguments (prevent unused arguments)
+    "B",      # flake8-bugbear (miscellaneous best practices to avoid bugs)
+    "C4",     # flake8-comprehensions (best practices for comprehensions)
+    "ICN",    # flake8-import-conventions (enforce import conventions)
+    "INP",    # flake8-no-pep420 (prevent use of PEP420, i.e. implicit name spaces)
+    "ISC",    # flake8-implicit-str-concat (conventions for concatenating long strings)
+    "LOG",    # flake8-logging
+    "NPY",    # numpy-specific rules
+    "PGH",    # pygrep-hooks (ensure appropriate usage of noqa and type-ignore)
+    "PTH",    # flake8-use-pathlib (enforce using Pathlib instead of os)
+    "SLF",    # flake8-self (prevent using private class members outside class)
+    "SLOT",   # flake8-slots (require __slots__ for immutable classes)
+    "T20",    # flake8-print (prevent print statements in code)
+    "TRY",    # tryceratops (best practices for try/except blocks)
+    "UP",     # pyupgrade (simplified syntax allowed by newer Python versions)
+    "YTT",    # flake8-2020 (prevent some specific gotchas from sys.version)
 ]
 ignore = [
+    "D100", # missing docstring in public module
     "E741", # ambiguous variable name (O/0, l/I, etc.)
+    "UP008", # use super() instead of super(class, self). no harm being explicit
+    "UP015", # unnecessary open(file, "r"). no harm being explicit
+    "TRY003", # prevents custom exception messages not defined in exception itself.
+    "ISC001", # single line implicit string concatenation. formatter recommends ignoring this.
 ]
 
 [tool.ruff.lint.pydocstyle]
 convention = "numpy"
 
+[tool.ruff.lint.pycodestyle]
+# Formatter autofixes to 88, but error thrown only above 100
+# This is primarily relevant for docstrings
+max-line-length = 100 
+
 [tool.ruff.lint.flake8-annotations]
 ignore-fully-untyped = true  # Turn of annotation checking for fully untyped code
 
-[tool.ruff.lint.extend-per-file-ignores]
+[tool.ruff.lint.per-file-ignores]
 "jwst/**/__init__.py" = [
     "E402", # module-level import not at top of file
     "F401", # unused import
     "F403", # import with *
     "F405", # ambiguous import from *
 ]
-"jwst/fits_generator/*" = [
-    "F821", # undefined name
-]
-"jwst/fits_generator/tests/*" = [
-    "F841", # unused variable
+"**/tests/__init__.py" = [
+    "D104", # missing docstring in public package
 ]
 "jwst/associations/tests*" = [
     "F841", # unused variable
 ]
-"jwst/ramp_fitting/tests/*" = [
-    "E402", # module-level import not at top of file
-]
+"jwst/step.py" = ["D", "N", "A", "ARG", "B", "C4", "ICN", "INP", "ISC", "LOG", "NPY", "PGH", "PTH", "SLF", "SLOT", "T20", "TRY", "UP", "YTT", "E501"]
```

Comment:
I suggest we split the ruff config into a separate file (see [the example in astropy](https://github.com/astropy/astropy/blob/main/.ruff.toml)) to avoid cluttering the `pyproject.toml`.

Reply:
Done, let me know how it looks